### ISSUE TITLE
Re-dowloading files deleted in Finder

### ISSUE THREAD

**Issue by jdawgesq** [rmcrackan/Libation#948]
Hello,

I started backing up my library with Audible, and for the first 50 books or so, I failed to check the box to split the downloads into chapters. The rest of my library downloaded with chapters, so that's great. After by library finished backing up, using Finder, I sorted and all of the downlads by date added and deleted all of those who had been downloaded as one single file.

When I rescanned my library in libation, the app appears to scan its db log without cross-referencing with the downloads folder and flag any missing files.

Is there a way to get Libation to scan the dowloads folder, compare it to the db, flag missing files and re-download them without having to start over completely and re-download the 500+ books in my library?

**Comment by rmcrackan** [comment]
tl;dr: Maybe.

The safest thing would be to manually go to each book which is not fixed and either right-click > Re-download this audiobook (starts the new download immediately) -or- rt-clk > Set download status to not-downloaded (starts the new download the next time you backup from the Liberate menu).

If you're feeling brave, you can backup the database file (LibationContext.db). Use the "Visible Books" menu to set all book statuses to not-downloaded. Then use Import > Locate Audiobooks. This only works if the audio file or its folder contains the original audible id (and if I didn't screw this feature up - I haven't tested in a long time).

If all goes right, not-downloaded will make them all appear as though they need to be downloaded again, then Locate Audiobooks will find the ids of books you already fixed and change them back to 'downloaded', thus only the to-be-fixed books will be picked up the next time you liberate.

That's a lot of if's and I would for sure never try this without backing up that database file first.

**Comment by jdawgesq** [comment]
Thanks for your quick reply. I really wish I’d kept a log of the books I
deleted in Finder.

The “locate books” function in the app appears to be a one-way street: it
seems to look for books that aren’t already in your  Libation db and add
them to the db, not the other way around. I’m trying to automate looking
for books in the db who don’t have matching files in the downloads folder,
and add them to the download queue.
